src/io.jl

```diff
@@ -81,6 +74,21 @@ function create_parameters_and_sets_from_file(input_folder::AbstractString)
     return params, sets
 end
 
+"""
+    get_df(path, file_name, schema)
+
+Reads the csv with file_name at location path, with specified number of header rows.
+Then validates the data using the schema.
+The silent argument is used in validate_df to optionally silence error messages.
+"""
+function get_df(path, file_name, header_rows, data_schema, silent = false)
+    csv_name = joinpath(path, file_name)
+    df = CSV.read(csv_name, DataFrames.DataFrame; header = header_rows)
+    validate_df(df, data_schema; fname = file_name, silent = silent)
```

Comment:
I think you misunderstood me a bit.  There is a common pattern (in many programming languages) when writing wrappers, you accept variable number of arguments (often called "varargs").  Then you transparently pass it on to the underlying function without altering it.  This way, you expose the configuration options of the underlying function to the caller of the wrapper without any special argument handling.

In our case, `CSV.read` has lots of options, and we don't know when we might need one, e.g. you get a CSV that's is  delimited by a semicolon instead of a comma, or maybe the first 3 lines have header info not first 2, etc.  Would we keep changing the wrapper function for every situation?

So instead we handle it such that this works:
- inside `create_parameters_and_sets_from_file`, we call `get_df` like this:
  ```julia
  df = get_df(dir, fname, schema_struct; header = 2, delim = ",")
  ```
- `get_df` looks something like this:
   ```julia
  function get_df(path, file_name, schema; csvargs...)
      # stuff
      df = CSV.read(...; csvargs...)
      validate_df(df, schema; fname = file_name)  # we don't want silent here, so default `false` is fine
      return df
  end
  ```

Reply:
Ah okay! Thanks I was thinking it was becoming a bit verbose. I think I incorporated your changes correctly.